flows/boundary.py

```diff
@@ -1289,9 +1280,17 @@ class DataPoint(TypedDict):
         id: VespaDataId
         fields: dict[str, Any]
 
-    def _to_data(text_block_id: TextBlockId, concepts: list[VespaConcept]) -> DataPoint:
-        document_passage_id = text_blocks[text_block_id][0]
-        document_passage = text_blocks[text_block_id][1]
+    def _to_data(
+        text_block_id: TextBlockId, concepts: list[VespaConcept]
+    ) -> DataPoint | None:
+        try:
+            document_passage_id = text_blocks[text_block_id][0]
+            document_passage = text_blocks[text_block_id][1]
+        except KeyError:
+            logger.error(
+                f"document passage `{text_block_id}` not found in Vespa for document `{document_import_id}`"
+            )
+            return None
```

Comment:
Catching key errors after seeing the issue in staging runs. 

I believe this is when we have text blocks from the labelled passage array that aren't yielded from vespa.